staking/programs/staking/src/state/vesting.rs

```diff
@@ -629,4 +639,147 @@ pub mod tests {
             None
         );
     }
+
+    #[quickcheck]
+    fn test_split_with_args(transferred: u64, total: u64, initial_balance: u64) -> TestResult {
+        if transferred > total || total == 0 {
+            return TestResult::discard();
+        }
+
+        let schedule = VestingSchedule::FullyVested;
+        let (remaining_schedule, transferred_schedule) = schedule
+            .split_vesting_schedule(total - transferred, transferred, total)
+            .unwrap();
+
+        assert_eq!(remaining_schedule, VestingSchedule::FullyVested);
+        assert_eq!(transferred_schedule, VestingSchedule::FullyVested);
+
+        let schedule = PeriodicVesting {
+            initial_balance,
+            // all of these fields should be preserved in the result
+            start_date: 203,
+            period_duration: 100,
+            num_periods: 12,
+        };
+        let (remaining_schedule, transferred_schedule) = schedule
+            .split_vesting_schedule(total - transferred, transferred, total)
+            .unwrap();
+
+        match (remaining_schedule, transferred_schedule) {
+            (
+                PeriodicVesting {
+                    initial_balance: r, ..
```

Comment:
I'd like to see a test that for an arbitrary t 
if `schedule.get_unvested_balance(t) <= total` 
`transfered_schedule.get_unvested_balance(t) <= transferred`
and  `transfered_schedule.get_unvested_balance(t) <= total - transferred`